src/WebJobs.Script.WebHost/Global.asax.cs

```diff
@@ -12,13 +15,36 @@ public class WebApiApplication : System.Web.HttpApplication
     {
         protected void Application_Start()
         {
+            var settingsManager = ScriptSettingsManager.Instance;
+            var webHostSettings = WebHostSettings.CreateDefault(settingsManager);
+
+            VerifyAndEnableShadowCopy(webHostSettings);
+
             using (var metricsLogger = new WebHostMetricsLogger())
             using (metricsLogger.LatencyEvent(MetricEventNames.ApplicationStartLatency))
             {
-                GlobalConfiguration.Configure(c => WebApiConfig.Initialize(c));
+                GlobalConfiguration.Configure(c => WebApiConfig.Initialize(c, settingsManager, webHostSettings));
             }
         }
 
+        private static void VerifyAndEnableShadowCopy(WebHostSettings webHostSettings)
+        {
+            if (!FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableShadowCopy))
+            {
+                string currentShadowCopyDirectories = AppDomain.CurrentDomain.SetupInformation.ShadowCopyDirectories;
+                string shadowCopyPath = GetShadowCopyPath(currentShadowCopyDirectories, webHostSettings.ScriptPath);
+
+#pragma warning disable CS0618
+                AppDomain.CurrentDomain.SetShadowCopyPath(shadowCopyPath);
+#pragma warning restore CS0618
+            }
+        }
+
+        internal static string GetShadowCopyPath(string currentShadowCopyDirectories, string scriptPath)
+        {
+            return string.Join(";", new[] { currentShadowCopyDirectories, scriptPath }.Where(s => !string.IsNullOrEmpty(s)));
```

Comment:
We could... just trying to simplify the logic here as we don't want to have a delimiter in the case where we have a single path (current is not set) or no paths (which shouldn't really change)

I actually had this a bit simpler (adding the delimiter if current was set) before, but this now accounts for a scenario where scriptPath wouldn't be set, which truly won't be the case, but this piece of code just takes that as a parameter. This change happened after testing this in isolation, but I'm fine switching it back if this is not as readable.